qiime2/core/cache.py

```diff
@@ -531,34 +883,120 @@ def _deallocate(self, symlink):
 
     @property
     def data(self):
+        """The directory in the cache that stores the data.
+        """
         return self.path / 'data'
 
+    def get_data(self):
+        """Returns a set of all data in the cache.
+
+        Returns
+        -------
+        set[str]
+            All of the data in the cache in the form of the top level
+            directories which will be the uuids of the artifacts.
+        """
+        return set(os.listdir(self.data))
+
     @property
     def keys(self):
+        """The directory in the cache that stores the keys.
+        """
         return self.path / 'keys'
 
+    def get_keys(self):
+        """Returns a set of all keys in the cache.
+
+        Returns
+        -------
+        set[str]
+            All of the keys in the cache. Just the names now what they refer
+            to.
+        """
+        return set(os.listdir(self.keys))
+
     @property
     def lockfile(self):
+        """The path to the flufl lock file.
+        """
         return self.path / 'LOCK'
 
     @property
     def pools(self):
+        """The directory in the cache that stores the named pools.
+        """
         return self.path / 'pools'
 
+    def get_pools(self):
+        """Returns a set of all pools in the cache.
+
+        Returns
+        -------
+        set[str]
+            The names of all of the named pools in the cache.
+        """
+        return set(os.listdir(self.pools))
+
     @property
     def processes(self):
+        """The directory in the cache that stores the process pools.
+        """
         return self.path / 'processes'
 
+    def get_processes(self):
+        """Returns a set of all process pools in the cache.
+
+        Returns
+        -------
+        set[str]
+            The names of all of the process pools in the cache.
+        """
+        return set(os.listdir(self.processes))
+
     @property
     def version(self):
+        """The path to the version file.
+        """
         return self.path / 'VERSION'
 
 
-# Assume we will make this its own class for now
 class Pool:
+    """Pools are folders in the cache that contain many symlinks to many
+    different piece of data. There are two types of pool:
+
+    **Process Pools:** These pools have names of the form
+    <pid>-<create-time>@<uname> based on the process that created them. They
```

Comment:
```suggestion
    <process-id>-<process-create-time>@<user> based on the process that created them. They
```

just to make it consistent with the env-var style above.